### PR TITLE
Fix - iOS NativeUserLocation puckBearing logic typo and puck image updates not being applied properly

### PR DESCRIPTION
**Pull request by mysport12**
## Description

Fixes:

- Incorrect puckBearing name in switch case
- Resolved warnings in Xcode
- Puck images not being fetched after initial addToMap
- Puck images not being correctly applied when changed on the fly

## Checklist

- [x] I've read `CONTRIBUTING.md`
- [ ] I updated the doc/other generated code with running `yarn generate` in the root folder
- [ ] I have tested the new feature on `/example` app.
  - [ ] In V11 mode/ios
  - [ ] In New Architecture mode/ios
  - [ ] In V11 mode/android
  - [ ] In New Architecture mode/android
- [ ] I added/updated a sample - if a new feature was implemented (`/example`)

## Component to reprocuce the issue you're fixing

```jsx
<Mapbox.MapView>
  <Mapbox.Images
          images={{ mapNavPuck: media.mapNavigationPuck }}
          onImageMissing={(imageKey: string) => console.log('=> on image missing', imageKey)}
        />
  <Mapbox.LocationPuck
            puckBearing={
              USER_TRACKING_MODES[userTrackingMode].key === 'followCourse'
                ? 'course'
                : 'heading'
            }
            puckBearingEnabled={true}
            topImage={
              USER_TRACKING_MODES[userTrackingMode].key === 'followCourse' ? 'mapNavPuck' : undefined
            }
          />
</Mapbox.MapView>
```

### CODE REVIEW

File: ios/RNMBX/RNMBXNativeUserLocation.swift [hunk 1]
```diff
@@ -63,12 +63,12 @@ public class RNMBXNativeUserLocation: UIView, RNMBXMapComponent {
...
       case "heading":
         _puckBearing = .heading
-      case "coures":
+      case "course":
         _puckBearing = .course
       case nil:
         _puckBearing = nil
       default:
-        Logger.error("RNMBXNativeUserLocation puckBearing is uncrecognized: \(value)")
+        Logger.error("RNMBXNativeUserLocation puckBearing is uncrecognized: \(String(describing: value))")
```
Comment (mfazekas) [inline ios/RNMBX/RNMBXNativeUserLocation.swift:71]:
```suggestion
        Logger.error("RNMBXNativeUserLocation puckBearing is uncrecognized: \(optional: value)")
```

File: ios/RNMBX/RNMBXNativeUserLocation.swift [hunk 2]
```diff
@@ -87,18 +87,8 @@ public class RNMBXNativeUserLocation: UIView, RNMBXMapComponent {
 
   func imageNameUpdated(_ image: PuckImagePart, _ name: String?) {
     imageNames[image] = name
-    if let subscription = subscriptions[image] {
-      subscription.cancel()
-      subscriptions.removeValue(forKey: image)
-    }
```
Comment (mfazekas) [inline ios/RNMBX/RNMBXNativeUserLocation.swift:93]: If name changed we still need to unsuscribe from old image don't we?

File: ios/RNMBX/RNMBXNativeUserLocation.swift [hunk 3]
```diff
@@ -132,11 +122,11 @@ public class RNMBXNativeUserLocation: UIView, RNMBXMapComponent {
         return Value.constant(0.0)
       }
     default:
-      Logger.error("toDoubleValue: \(name): has unknown type: \(type(of:value)) \(value) ")
+      Logger.error("toDoubleValue: \(name): has unknown type: \(type(of:value)) \(String(describing: value)) ")
```
Comment (mfazekas) [inline ios/RNMBX/RNMBXNativeUserLocation.swift:125]:
```suggestion
      Logger.error("toDoubleValue: \(name): has unknown type: \(type(of:value)) \(optional: value) ")
```

**Review by mfazekas — CHANGES_REQUESTED** [review]

File: ios/RNMBX/RNMBXNativeUserLocation.swift (hunk 2 above)
Comment (mysport12) [reply ios/RNMBX/RNMBXNativeUserLocation.swift:93]: All subscriptions are removed in the _fetchImages function (line 266), and then new ones are created for the updated imageNames array (block starting on line 268)

File: ios/RNMBX/RNMBXNativeUserLocation.swift (hunk 3 above)
Comment (mysport12) [reply ios/RNMBX/RNMBXNativeUserLocation.swift:125]: These changes were proposed by Xcode not something I changed just because. Fine to remove/revert

**Pull request merged**